lib/srv/sess.go

```diff
@@ -334,8 +341,8 @@ func (r *sessionRegistry) PartyForConnection(sconn *ssh.ServerConn) *party {
 
 	for _, session := range r.sessions {
 		session.Lock()
+		defer session.Unlock()
```

Comment:
this is the correct intent: the loop is actually a simple slice range, the number of sessions is tiny and each one needs to be locked to access it's "parties" slice.